activerecord/lib/active_record/secure_token.rb

```diff
@@ -23,11 +23,14 @@ module ClassMethods
       # Note that it's still possible to generate a race condition in the database in the same way that
       # <tt>validates_uniqueness_of</tt> can. You're encouraged to add a unique index in the database to deal
       # with this even more unlikely scenario.
```

Comment:
We also need to add documentation for your change. Feel to use your example from the use case example as a base (but maybe slim it down a little).

The documentation needs to say something like:

```ruby
# A secure token can also be created under certain circumstances. Say a user should
# only have an invitation token if they where invited:
#
#   your code indented by three spaces
#
# The secure token creation supports all the options a `before_create` does.
```